pkg/state/state_ofed.go

```diff
@@ -195,3 +204,32 @@ func (s *stateOFED) getManifestObjects(
 	log.V(consts.LogLevelDebug).Info("Rendered", "objects:", objs)
 	return objs, nil
 }
+
+// getMofedDriverImageName generates MOFED driver image name based on the driver version specified in CR
+// TODO(adrianc): in Network-Operator v1.5.0, we should just use the new naming scheme
+func (s *stateOFED) getMofedDriverImageName(cr *mellanoxv1alpha1.NicClusterPolicy,
```

Comment:
nit: reduce typing
```
// getMofedDriverImageName generates MOFED driver image name based on the driver version specified in CR
// TODO(adrianc): in Network-Operator v1.5.0, we should just use the new naming scheme
func (s *stateOFED) getMofedDriverImageName(cr *mellanoxv1alpha1.NicClusterPolicy,
	nodeAttr map[nodeinfo.AttributeType]string) string {
	// format: <repo>/<image-name>:<driver-version>-<os-name><os-ver>-<cpu-arch>
	// e.x: nvcr.io/nvidia/mellanox/mofed:5.7-0.0.8.0-ubuntu20.04-amd64.
	mofedImageNameTmpl := "%s/%s:%s-%s%s-%s"
	if cr.Spec.OFEDDriver.Version < newMofedImageFormatVersion  {
		// format: <repo>/<image-name>-<driver-version>:<os-name><os-ver>-<cpu-arch>
		// e.x: nvcr.io/nvidia/mellanox/mofed-5.6-1.0.3.3:ubuntu20.04-amd64
		mofedImageNameTmpl = "%s/%s-%s:%s%s-%s"
	}
	return fmt.Sprintf(mofedImageNameTmpl,
		cr.Spec.OFEDDriver.Repository,
		cr.Spec.OFEDDriver.Image,
		cr.Spec.OFEDDriver.Version,
		nodeAttr[nodeinfo.AttrTypeOSName],
		nodeAttr[nodeinfo.AttrTypeOSVer],
		nodeAttr[nodeinfo.AttrTypeCPUArch])
}
```

Reply:
sure, ill simplify

Reply:
Done. LMK if thats OK.

this method now uses Masterminds/semver to parse the version so need to handle possible errors.
current flow will fallback to new image format and log in debug.

which will eventually lead to mofed daemonset to fail to pull image most likely like today.

Reply:
looks good, thanks